contracts/contracts/gateway/router/CheckpointingFacet.sol

```diff
@@ -58,66 +42,14 @@ contract CheckpointingFacet is GatewayActorModifiers {
-    /// @param msgs - the full messages batch
-    /// @param activity - the full activity rollup
-    function createBottomUpCheckpoint(
-        BottomUpCheckpoint calldata checkpoint,
-        bytes32 membershipRootHash,
-        uint256 membershipWeight,
-        IpcEnvelope[] calldata msgs,
-        FullActivityRollup calldata activity
-    ) external systemActorOnly {
-        if (LibGateway.bottomUpCheckpointExists(checkpoint.blockHeight)) {
-            revert CheckpointAlreadyExists();
-        }
-
-        LibQuorum.createQuorumInfo({
-            self: s.checkpointQuorumMap,
-            objHeight: checkpoint.blockHeight,
-            objHash: keccak256(abi.encode(checkpoint)),
-            membershipRootHash: membershipRootHash,
-            membershipWeight: membershipWeight,
-            majorityPercentage: s.majorityPercentage
-        });
-
-        LibGateway.storeBottomUpCheckpoint(checkpoint);
-
-        emit BottomUpBatchRecorded(uint64(checkpoint.blockHeight), msgs);
-        emit ActivityRollupRecorded(uint64(checkpoint.blockHeight), activity);
-    }
-
-    /// @notice checks whether the provided checkpoint signature for the block at height `height` is valid and accumulates that it
-    /// @dev If adding the signature leads to reaching the threshold, then the checkpoint is removed from `incompleteCheckpoints`
-    /// @param height - the height of the block in the checkpoint
-    /// @param membershipProof - a Merkle proof that the validator was in the membership at height `height` with weight `weight`
-    /// @param weight - the weight of the validator
-    /// @param signature - the signature of the checkpoint
-    function addCheckpointSignature(
-        uint256 height,
-        bytes32[] memory membershipProof,
-        uint256 weight,
-        bytes memory signature
-    ) external {
-        // check if the checkpoint was already pruned before getting checkpoint
-        // and triggering the signature
-        LibQuorum.isHeightAlreadyProcessed(s.checkpointQuorumMap, height);
-
-        // slither-disable-next-line unused-return
-        (bool exists, ) = LibGateway.getBottomUpCheckpoint(height);
-        if (!exists) {
-            revert CheckpointNotCreated();
-        }
-        LibQuorum.addQuorumSignature({
-            self: s.checkpointQuorumMap,
-            height: height,
-            membershipProof: membershipProof,
-            weight: weight,
-            signature: signature
-        });
+    /// @notice submit a verified checkpoint in the gateway to trigger side-effects.
+    /// @dev this method is called by the corresponding subnet actor.
+    ///     Called from a subnet actor if the checkpoint is cryptographically valid.
+    /// @param checkpoint The bottom-up checkpoint to be committed.
+    function commitCheckpoint(BottomUpCheckpoint calldata checkpoint) external systemActorOnly {
+        emit StateCommitmentCreated(uint64(block.number), checkpoint.commitment);
```

Comment:
What about checking for `error CheckpointAlreadyExists`?

Reply:
I removed it as this is part of the system call and the caller has height checked. But I guess there is no harm adding this check as it makes no assumptions. Adding back.